### ISSUE TITLE
Thread.new should inherit the interrupt mask of the calling thread

### ISSUE THREAD

**Issue by larskanis** [truffleruby/truffleruby#1447]
This was recently fixed in JRuby here: https://github.com/jruby/jruby/issues/5314 . It would be nice to get this fixed in Truffleruby as well. Until then I use the following [workaround](https://github.com/larskanis/eventbox/commit/a0f772f73fe8690286311cb353502617f4326994) for [Eventbox](https://github.com/larskanis/eventbox).

**Comment by chrisseaton** [comment]
The 'interrupt mask' is the thing that `handle_interrupt` sets, is that right? It's not about signal masks of the underlying threads or anything like that?

**Comment by larskanis** [comment]
Exactly, the thing that `handle_interrupt` sets. Since it can not mask anything other than Exceptions for now, it has nothing to do with signals. See also my comment here: https://github.com/jruby/jruby/issues/5314#issuecomment-426713442

**Comment by eregon** [comment]
FWIW in the context of the timeout gem I think inheriting the interrupt mask might actually be a bug on CRuby, and I had to do the reverse workaround to not inherit it.
In your case, could you explain how is the inheritance of the interrupt mask useful/necessary?
